src/TriggerBinding/SqlTriggerListener.cs

```diff
@@ -0,0 +1,364 @@
+            _ = executor ?? throw new ArgumentNullException(nameof(executor));
+            _ = logger ?? throw new ArgumentNullException(nameof(logger));
+
+            this._connectionString = connectionString;
+            this._userTable = new SqlObject(tableName);
+            this._userFunctionId = userFunctionId;
+            this._executor = executor;
+            this._logger = logger;
+            this._state = State.NotInitialized;
+        }
+
+        /// <summary>
+        /// Stops the listener which stops checking for changes on the user's table.
+        /// </summary>
+        public void Cancel()
+        {
+            this.StopAsync(CancellationToken.None).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Disposes resources held by the listener to poll for changes.
+        /// </summary>
+        public void Dispose()
+        {
+            this.StopAsync(CancellationToken.None).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// Starts the listener if it has not yet been started, which starts polling for changes on the user's table.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token</param>
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            if (this._state == State.NotInitialized)
+            {
+                using (var connection = new SqlConnection(this._connectionString))
+                {
+                    await connection.OpenAsync(cancellationToken);
+
+                    int userTableId = await this.GetUserTableIdAsync(connection, cancellationToken);
+                    IReadOnlyList<(string name, string type)> primaryKeyColumns = await this.GetPrimaryKeyColumnsAsync(connection, cancellationToken);
+                    IReadOnlyList<string> userTableColumns = await this.GetUserTableColumnsAsync(connection, cancellationToken);
+
+                    string workerTableName = string.Format(CultureInfo.InvariantCulture, SqlTriggerConstants.WorkerTableNameFormat, $"{this._userFunctionId}_{userTableId}");
+
+                    using (SqlTransaction transaction = connection.BeginTransaction(System.Data.IsolationLevel.RepeatableRead))
+                    {
+                        await CreateSchemaAsync(connection, transaction, cancellationToken);
+                        await CreateGlobalStateTableAsync(connection, transaction, cancellationToken);
+                        await this.InsertGlobalStateTableRowAsync(connection, transaction, userTableId, cancellationToken);
+                        await CreateWorkerTablesAsync(connection, transaction, workerTableName, primaryKeyColumns, cancellationToken);
+                        transaction.Commit();
+                    }
+
+                    // TODO: Check if passing cancellation token would be beneficial.
```

Comment:
I would assume we need to do this to properly handle cancellations from the host - is this something you're planning on fixing?